rust/rsc/src/rsc/blob.rs

```diff
@@ -0,0 +1,71 @@
+use crate::config::RSCConfig;
+use crate::types::GetUploadUrlResponse;
+use axum::{extract::Multipart, Json};
+use sea_orm::DatabaseConnection;
+use std::sync::Arc;
+use tracing;
+
+#[tracing::instrument]
+pub async fn get_upload_url(
+    _conn: Arc<DatabaseConnection>,
+    config: Arc<RSCConfig>,
+) -> Json<GetUploadUrlResponse> {
+    let url = config.server_addr.clone() + "/blob";
+    Json(GetUploadUrlResponse { url })
+}
+
+pub async fn create_blob(mut multipart: Multipart, config: Arc<RSCConfig>) -> hyper::StatusCode {
+    let mut hash: Option<String> = None;
+    let mut contents: Option<axum::body::Bytes> = None;
+
+    let exhausted_without_err = loop {
+        let Ok(field) = multipart.next_field().await else {
+            break false;
+        };
+
+        // if field is None, then we have exhausted the input.
+        let Some(field) = field else {
+            break true;
+        };
+
+        if field.name().is_none() {
+            break false;
+        }
+        let name = field.name().unwrap().to_string();
+        let Ok(data) = field.bytes().await else {
+            break false;
+        };
+
+        if name == "hash" {
+            let key = String::from_utf8(data.to_ascii_lowercase()).unwrap();
+            hash = Some(key);
+        }
```

Comment:
I think the server should be the one doing the hashing, not the client? I can't remember what we decided but my current thinking is that the server should be the one hashing the blob

Reply:
Sent in a DM yesterday :)

For the MVP I was thinking we should just have the client hash because 1) it already knows the hash 2) we trust all the uploaders 3) I think server rehashing messes with streaming

Reply:
pushing streaming off to later PR per DMs